src/main/java/org/spongepowered/api/util/ban/Ban.java

```diff
@@ -83,21 +84,42 @@ static Ban of(org.spongepowered.api.entity.living.player.User user, Text.Literal
      *
      * @return The reason specified for the ban.
      */
-    Text.Literal getReason();
+    Text getReason();
 
     /**
-     * Gets the start date of the ban.
+     * Gets the creation date of the ban.
+     *
+     * <p>Note that this {@link Date} has no effect on whether or not a ban is
+     * active. Any ban for which {@link BanService#hasBan(Ban)} returns
+     * <code>true</code> will be used (when checking if a player can join,
+     * for example), regardless of its creation date.</p>
      *
      * @return Creation date of the ban
      */
-    Date getStartDate();
+    Date getCreationDate();
+
+    /**
+     * Gets the source that created this ban, if available
+     *
+     * <p>Depending on the implementation, the returned {@link Text}
+     * may represent a {@link CommandSource}. {@link #getBanCommandSource()} B()} can be
+     * used to attempt to convert the source to a {@link CommandSource}.</p>
+     *
+     * @return the source of this ban, if available
+     */
+    Optional<Text> getBanSource();
```

Comment:
Having a `Text` ban source is a little wierd. Whats the issue with having this be a `CommandSource` is their any case where the source is present and is not a command source?